macros/REST_OpenInputFile.C

```diff
@@ -1,40 +1,44 @@
-TRestRun* run0 = nullptr;
-TRestAnalysisTree* ana_tree0 = nullptr;
-TTree* ev_tree0 = nullptr;
-TRestDataSet* dSet0 = nullptr;
-std::map<std::string, TRestMetadata*> metadata0;
+TRestRun* run = nullptr;
+TRestAnalysisTree* ana_tree = nullptr;
+TTree* ev_tree = nullptr;
+TRestEvent* ev = nullptr;
+TRestDataSet* dSet = nullptr;
+std::map<std::string, TRestMetadata*> metadata;
 
 void REST_OpenInputFile(const std::string& fileName) {
     if (TRestTools::isRunFile(fileName)) {
         printf("\n%s\n", "REST processed file identified. It contains a valid TRestRun.");
-        run0 = new TRestRun(fileName);
-        printf("\nAttaching TRestRun %s as run0...\n", fileName.c_str());
-        ana_tree0 = run0->GetAnalysisTree();
-        printf("\nAttaching TRestAnalysisTree as ana_tree0...\n");
-        ev_tree0 = run0->GetEventTree();
-        printf("\nAttaching event tree as ev_tree0...\n");
+        run = new TRestRun(fileName);
+        printf("\nAttaching TRestRun %s as run...\n", fileName.c_str());
+        ana_tree = run->GetAnalysisTree();
+        printf("\nAttaching TRestAnalysisTree as ana_tree...\n");
+        ev_tree = run->GetEventTree();
+        printf("\nAttaching event tree as ev_tree...\n");
+        ev = run->GetInputEvent();
```

Comment:
In the previous `restRoot` version we were creating an instance of the specific event type. That would allow us to access specific methods. Of course, we will still have access to `PrintEvent`, `DrawEvent`, and other common methods.

```
                string eventType = runTmp->GetInputEvent()->ClassName();

                printf("Attaching event %s as ev%i...\n", eventType.c_str(), nFile);
                string evcmd = Form("%s* ev%i = (%s*)%s;", eventType.c_str(), nFile, eventType.c_str(),
                                    (PTR_ADDR_PREFIX + ToString(runTmp->GetInputEvent())).c_str());
```

But just wondering if there is a solution for that.

Reply:
I am afraid there is no solution for that, before it was possible because it was using `gROOT->ProcessLine(...)` but I think we shouldn't be using that. Alternatively, you can do in the root terminal `auto rawEvent = static_cast<TRestRawSignalEvent *> (ev)`